src/Language/Parser.php

```diff
@@ -62,65 +62,57 @@ class Parser implements ParserInterface
     protected $lexer;
 
     /**
-     * Given a GraphQL source, parses it into a Document.
-     * Throws GraphQLError if a syntax error is encountered.
+     * @param string $name
+     * @param array  $arguments
+     * @return mixed
      *
-     * @inheritdoc
-     * @throws SyntaxErrorException
-     * @throws \ReflectionException
      * @throws InvariantException
+     * @throws SyntaxErrorException
      */
-    public function parse($source, array $options = []): DocumentNode
+    public function __call(string $name, array $arguments)
```

Comment:
Actually, now that i thought about it some more, i do think that using a magic method here is not the best way to go. It does solve the issue with very few lines of code, but on the other hand makes the code less clean.

Reply:
I would normally agree, however in this case we would need 30+ parse methods, so I opted for this approach instead. This is also not really part of the public api so I think it’s acceptable.

Reply:
@spawnia thoughts? Can we go with this solution now, as I’d like to keep it simple. We can change it later if need be.